File: Data Structures & Algorithms/concatenation-of-array/submission-0.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int length = nums.length;
        int[] newArray = new int[length * 2]; // all 0's?
        for (int i = 0; i < nums.length; i++) {
            newArray[i] = nums[i];
            newArray[i + length] = nums[i];
        }
        return newArray;
    }
}

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        nums.sort() # sort nums to avoid duplicates

        subsets = []

        def backtrack(chosen, remaining):
            if chosen not in subsets:
                subsets.append(chosen[:])
            for i in range(len(remaining)):
                chosen.append(remaining[i])
                backtrack(chosen, remaining[i+1:])
                chosen.pop()
        backtrack([], nums)
        return subsets